### ISSUE TITLE
Default Confidence 

### ISSUE THREAD

**Issue by chungyau97** [zisianw/FaceBoxes.PyTorch#24]
parser.add_argument('--confidence_threshold', default=0.05, type=float, help='confidence_threshold')

The default condifence in pytorch and caffe is different.

Is it too low in pytorch?

**Comment by zisianw** [comment]
Hi @chungyau97, may I know where in the caffe code is the default confidence different?

**Comment by chungyau97** [comment]
Correct me if I'm wrong.

Based on the caffe. I think it's set as 50% while in pytorch 0.05 is 5%? 
![image](https://user-images.githubusercontent.com/33013947/54168592-3ffa2d00-44aa-11e9-8cb9-0cef2dd71264.png)

**Comment by zisianw** [comment]
The 0.5 conf filtering is in [demo code](https://github.com/sfzhang15/FaceBoxes/blob/master/test/demo.py#L37) used for visualization, plotting bbox with conf score 0.5 and above. If you check other test files used for evaluation such as [afw_test.py](https://github.com/sfzhang15/FaceBoxes/blob/master/test/afw_test.py) there is no filtering as such. The scores from output are > 0.05 according to [deploy.prototxt](https://github.com/sfzhang15/FaceBoxes/blob/master/models/faceboxes/deploy.prototxt#L1776).

**Comment by chungyau97** [comment]
So if I want to plot bbox on different resolution of my own dataset images. I need to change the 0.05 to 0.5 for visualization purpose of result in the image right? 

Can you tell me more about the resize? This is my code that I play around with it 
![image](https://user-images.githubusercontent.com/33013947/54172126-f9abca80-44b7-11e9-8a48-5cda46efdbc7.png)

**Comment by zisianw** [comment]
You can change the threshold to any value depending on the conf score you wish to filter. “resize” is a parameter that resizes the image proportionally before putting them into the network. Larger input image could produce better results as small faces are enlarged, at the cost of longer inference time.

**Comment by chungyau97** [comment]
Correct me if I'm wrong.
So if the resize value is larger, the longer it take to inference an image and the higher the accuracy of getting faces correctly right? 

And I'm very curios with the pre-processing image mean subtraction (104, 117, 123) in testing and training. I can't find this in your research paper. Can you explain more in this?

**Comment by hsl20130659** [comment]
Did the Anchor densification strategy realize?

**Comment by zisianw** [comment]
@chungyau97 generally yes, you can run an experiment on your test image and check detection performance as well as the inference time. As for mean subtraction, it is a common preprocessing step. Courses or tutorial you find on the internet will explain in detailed.

**Comment by zisianw** [comment]
@hsl20130659 yes, it is implemented in this repo.

**Comment by hsl20130659** [comment]
I have some confusion in prior_box.py
s_kx = min_size / self.image_size[1]
s_ky = min_size / self.image_size[0]
dense_cx = [x*self.steps[k]/self.image_size[1] for x in [j+0, j+0.25, j+0.5, j+0.75]]
what do s_kx and s_ky mean? why do this?
And x*self.steps[k]/self.image_size[1] ,why do x mutiplies self.steps[k]/self.image_size[1] ?

**Comment by chungyau97** [comment]
@zisianw @hsl20130659 
out = net(img)
loc, conf, feature_map = out
#convert feature map from tensor to numpy
priorbox = PriorBox(cfg, feature_map , (im_height, im_width), phase='test')
#Find default boxes best match with ground truth?
priors = priorbox.forward()

**Comment by zisianw** [comment]
@hsl20130659, [`cx, cy, s_kx, s_ky]` are the center and scale of the anchor boxes. So `s_kx` and `s_ky` are the width and height of the anchor boxes normalized by the image size. `x` and `y` are one of the `[j+0, j+0.25, j+0.5, j+0.75]`, in your example which belongs to anchor size of 32x32, the combination of `dense_cx` and `dense_cy` makes up the center of the 16 anchors according to anchor densification.

**Comment by hsl20130659** [comment]
s_kx and s_ky are the width and height of the anchor boxes normalized by the image size，why do normalization by image size ？

**Comment by zisianw** [comment]
So that scale of anchor boxes are relative to input image.

**Comment by chungyau97** [comment]
@zisianw 

> @hsl20130659, [`cx, cy, s_kx, s_ky]` are the center and scale of the anchor boxes. So `s_kx` and `s_ky` are the width and height of the anchor boxes normalized by the image size. `x` and `y` are one of the `[j+0, j+0.25, j+0.5, j+0.75]`, in your example which belongs to anchor size of 32x32, the combination of `dense_cx` and `dense_cy` makes up the center of the 16 anchors according to anchor densification.

So priorbox is basically normalize the size of anchor box right?

**Comment by zisianw** [comment]
We used the term prior box and anchor box interchangeably. They mean the same thing. It is normalized by input image size, which in our training is 1024.
